crates/generator/src/syscalls/mod.rs

```diff
@@ -593,8 +599,7 @@ impl<'a, S: State, C: ChainView> L2Syscalls<'a, S, C> {
             addr += 1;
         }
 
-        let s = String::from_utf8(buffer).map_err(|_| VMError::ParseError)?;
-        log::debug!("[contract debug]: {}", s);
+        self.redir_log_handler.append_log(buffer);
```

Comment:
This seem to change the semantics of `output_debug`. Non-valid UTF-8 is now allowed. Is this intentional?

How about still convert to `String` here, and change `sink` to be a `String` too?

Reply:
makes sense